src/components/OpenSeadragonViewer.js

```diff
@@ -203,14 +204,14 @@ export class OpenSeadragonViewer extends Component {
   }
 
   /** */
-  addAllImageSources() {
+  addAllImageSources(zoomAfterAdd = true) {
     const { nonTiledImages, infoResponses } = this.props;
     Promise.all(
       infoResponses.map(infoResponse => this.addTileSource(infoResponse)),
       nonTiledImages.map(image => this.addNonTiledImage(image)),
     ).then(() => {
       if (infoResponses[0] || nonTiledImages[0]) {
-        this.zoomToWorld();
+        if (zoomAfterAdd) this.zoomToWorld();
```

Comment:
I struggled with figuring out a way to test this async code. We could do some refactoring to make that happen but not sure if we should do all of that.